jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java

```diff
@@ -1132,6 +1132,74 @@ public void testDeleteFailedJvm() {
+
+		CommandOutput commandOutput = mock(CommandOutput.class);
+		Jvm mockJvm = mock(Jvm.class);
+		ResourceGroup mockResourceGroup = mock(ResourceGroup.class);
+
+		when(mockJvm.getState()).thenReturn(JvmState.JVM_STOPPED);
+		when(mockJvm.getJvmName()).thenReturn("test-jvm-deploy-config");
+		when(mockJvm.getId()).thenReturn(new Identifier<Jvm>(111L));
+		when(mockJvm.getJdkMedia())
+				.thenReturn(new Media(1L, "test JDK media", MediaType.JDK, Paths.get("x:/test/archive/path.zip"),
+						Paths.get("x:/test-destination"), Paths.get("root-dir-destination")));
+		when(mockJvm.getTomcatMedia()).thenReturn(new Media(2L, "test Tomcat media", MediaType.TOMCAT,
+				Paths.get("./src/test/resources/binaries/apache-tomcat-test.zip"),
+				Paths.get("x:/test-destination-tomcat"), Paths.get("tomcat-root-dir-destination")));
+		when(commandOutput.getReturnCode()).thenReturn(new ExecReturnCode(0));
+		when(Config.mockJvmControlService.secureCopyFile(any(ControlJvmRequest.class), anyString(), anyString(),
+				anyString(), anyBoolean())).thenReturn(commandOutput);
+		when(Config.mockJvmControlService.executeCreateDirectoryCommand(any(Jvm.class), anyString()))
+				.thenReturn(commandOutput);
+		when(Config.mockJvmControlService.executeChangeFileModeCommand(any(Jvm.class), anyString(), anyString(),
+				anyString())).thenReturn(commandOutput);
+		when(Config.mockJvmControlService.controlJvm(
+				eq(ControlJvmRequestFactory.create(JvmControlOperation.DELETE_SERVICE, mockJvm)), any(User.class)))
+						.thenReturn(commandOutput);
+		when(Config.mockJvmControlService.controlJvm(
+				eq(ControlJvmRequestFactory.create(JvmControlOperation.DEPLOY_JVM_ARCHIVE, mockJvm)), any(User.class)))
+						.thenReturn(commandOutput);
+		when(Config.mockJvmControlService.controlJvm(
+				eq(ControlJvmRequestFactory.create(JvmControlOperation.INSTALL_SERVICE, mockJvm)), any(User.class)))
+						.thenReturn(commandOutput);
+		when(Config.mockJvmControlService.executeCheckFileExistsCommand(any(Jvm.class), anyString()))
+				.thenReturn(commandOutput);
+
+		when(Config.mockJvmPersistenceService.findJvmByExactName(anyString())).thenReturn(mockJvm);
+		when(Config.mockJvmPersistenceService.getJvmTemplate(anyString(), any(Identifier.class)))
+				.thenReturn("<server>some xml</server>");
+
+		when(Config.mockResourceService.generateResourceGroup()).thenReturn(mockResourceGroup);
+		when(Config.mockResourceService.generateResourceFile(anyString(), anyString(), any(ResourceGroup.class),
+				anyObject(), any(ResourceGeneratorType.class))).thenReturn("<server>some xml</server>");
+
+		final List<String> templateNames = new ArrayList<>();
+		templateNames.add("setenv.bat");
+		when(Config.mockJvmPersistenceService.getResourceTemplateNames(anyString())).thenReturn(templateNames);
+
+		Jvm response = jvmService.upgradeJDK(mockJvm.getJvmName(), Config.mockUser);
+		assertEquals(response.getJvmName(), mockJvm.getJvmName());
+
```

Comment:
Add verify() calls to check that the mocks are getting called as expected and that the interactions with the other services are occurring as you expected.

Example verify calls: https://github.com/cerner/jwala/blob/JDK_upgrade_enhancement/jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java#L137